[BUG] Unit test at `tests/datasets/test_datasets.py::test_settings_workflow` failing when running `pytest`

**Describe the bug**

As of the recent addition of the new `owner` role, it seems that a unit test is failing due to unknown reasons, most likely due to a residual test that's changing something unexpected, that's why we've decided it to skip `tests/datasets/test_datasets.py::test_settings_workflow` for the moment while we look for a solution.

**Stacktrace and Code to create the bug**

Simply running the whole test suite as `pytest` will eventually lead into this issue, but running `pytest tests/datasets/test_datasets.py::test_settings_workflow` won't fail.

<details>
<summary>How do I dropdown?</summary>
<pre>
FAILED tests/datasets/test_datasets.py::test_settings_workflow[settings_0-wrong_settings0] - argilla.client.sdk.commons.errors.UnauthorizedApiError: Argilla server returned an error with http status: 401
Error details: [{'code': 'argilla.api.errors::UnauthorizedError', 'params': {'detail': 'Could not validate credentials'}}]
</pre>
</details>

**Expected behavior**

The test should pass as it pass when ran as standalone.

**Environment:**
 - Argilla Version [e.g. 1.0.0]: 1.11.0.dev0
 - ElasticSearch Version [e.g. 7.10.2]: 7.17, 8, 8.5, 8.8
 - Docker Image (optional) [e.g. argilla:v1.0.0]: argilla:v1.11.0

--- a/tests/datasets/test_datasets.py
+++ b/tests/datasets/test_datasets.py
@@ -18,7 +18,8 @@ from argilla.client.client import Argilla
 from argilla.client.sdk.commons.errors import ForbiddenApiError
 from argilla.datasets import TextClassificationSettings, TokenClassificationSettings
 from argilla.server.contexts import accounts
-from argilla.server.security.model import User, WorkspaceUserCreate
+from argilla.server.models import User
+from argilla.server.security.model import WorkspaceUserCreate
 from sqlalchemy.orm import Session
 
 

--- a/tests/datasets/test_datasets.py
+++ b/tests/datasets/test_datasets.py
@@ -18,7 +18,7 @@ from argilla.client.client import Argilla
 from argilla.client.sdk.commons.errors import ForbiddenApiError
 from argilla.datasets import TextClassificationSettings, TokenClassificationSettings
 from argilla.server.contexts import accounts
-from argilla.server.security.model import WorkspaceUserCreate
+from argilla.server.security.model import User, WorkspaceUserCreate
 from sqlalchemy.orm import Session
 
 
@@ -35,8 +35,10 @@ from sqlalchemy.orm import Session
         ),
     ],
 )
-def test_settings_workflow(mocked_client, settings_, wrong_settings):
+def test_settings_workflow(argilla_user: User, settings_, wrong_settings):
     dataset = "test-dataset"
+
+    rg.init(api_key=argilla_user.api_key, workspace=argilla_user.username)
     workspace = rg.get_workspace()
 
     rg.delete(dataset)

